src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,135 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> tailNode = tail;
+        Node<T> newNode = new Node<>(tailNode, value, null);
+        tail = newNode;
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tailNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNode(index);
+        Node<T> newNode = new Node<T>(currentNode.prev, value, currentNode);
+        if (currentNode.prev == null) {
+            head = newNode;
+        } else {
+            currentNode.prev.next = newNode;
+        }
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T numbers : list) {
+            add(numbers);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> setNode = getNode(index);
+        T setResult = setNode.item;
+        setNode.item = value;
+        return setResult;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removeNode = getNode(index);
+        T removeValue = removeNode.item;
+        unlink(removeNode);
+        return removeValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> thisNode = head;
+        while (thisNode != null) {
+            if (thisNode.item == object || object != null && object.equals(thisNode.item)) {
+                unlink(thisNode);
+                return true;
+            }
+            thisNode = thisNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Something was wrong.");
```

Comment:
Something? Very informative))) May be you add info about the index and actual size?